Src/VimCore/CommandUtil.fs

```diff
@@ -944,8 +944,13 @@ type internal CommandUtil
 
     /// Get the appropriate register for the CommandData
     member x.GetRegister (commandData : CommandData) =
-        let name = 
+            match commandData.RegisterName with
+            | Some name -> Some (_registerMap.GetRegister name)
+            | None -> None
 
+    /// Get the appropriate register for the CommandData
+    member x.GetRegisterForPut (commandData : CommandData) =

```

Comment:
Consider if there  was a `SetRegisterValue` method here.  This method could take into account the `clipboard` option and update both the clipboard and unnamed register here.  This would be a nice pairing with `GetRegister` which has similar logic.